data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineTransformer.java

```diff
@@ -112,11 +119,32 @@ private void buildPipelineFromConfiguration(
             final PluginSetting sourceSetting = pipelineConfiguration.getSourcePluginSetting();
             final Optional<Source> pipelineSource = getSourceIfPipelineType(pipelineName, sourceSetting,
                     pipelineMap, pipelineConfigurationMap);
-            final Source source = pipelineSource.orElseGet(() ->
-                    pluginFactory.loadPlugin(Source.class, sourceSetting));
+            final Source source = pipelineSource.orElseGet(() -> {
+                try {
+                    return pluginFactory.loadPlugin(Source.class, sourceSetting);
+                } catch (Exception e) {
+                    final PluginError pluginError = PluginError.builder()
+                            .pipelineName(pipelineName)
```

Comment:
Should we include the "componentType?" That is, `source`, `processor`, `sink`, buffer`?

Reply:
Thanks for the catch! Will add that